src/ffi/authenticator.js

```diff
@@ -684,10 +698,52 @@ class Authenticator extends SafeLib {
   }
 
   _isAlreadyAuthorised(req) {
-    return this.getRegisteredApps()
-      .then((authorisedApps) =>
-        ((authorisedApps.filter((apps) =>
-          (lodash.isEqual(apps.app_info, req.app)))).length !== 0));
+    let containerLen = req.containers.length;
+    if (req.app_container) {
+      containerLen += 1;
+    }
+    let app = null;
+    return new Promise((resolve) => {
```

Comment:
Handle `reject`. Adding a try catch block is better.
Also the own_container condition is not validated